### DISCUSSION TITLE
Allow making multiple `pg_net` requests in parallel from a query

### DISCUSSION THREAD

**Discussion by nikodraca** [supabase/supabase#27183] [Feature Requests] [answered]
Hi team 👋 I'm wondering if you've considered some way of creating parallel requests based on a query response:
1. Query table and get rows
2. For each row, make a request in parallel where the payload is the row 

Maybe there's an existing way to do this but I could only find a way to make a _single_ request based on the results of many rows

**Comment by vickkhera** [comment, marked as answer]
You can call `net.http_post()` as a function in your select, and it will operate on every row that matched.

Assume you have a table called `tasks` and a column called `post_data` with the json for the endpoint you want to call with pg_net. Now you can submit all of them at once like this:

```
select net.http_post('http://example.com', post_data) from tasks;
```

They will all be added to the queue and executed upon completion of that statement (or the transaction if you are inside one).

**Reply by nikodraca** [reply]
Very cool, did not realize you could do this, thanks!
